File: Python/OneLang/Utils/ArrayHelper.py
from OneLangStdLib import *

class ArrayHelper:
    def __init__(self):
        pass
    
    @classmethod
    def sort_by(cls, items, key_selector):
        # @java-import java.util.Arrays
        # @java Arrays.sort(items, (a, b) -> keySelector.apply(a) - keySelector.apply(b));
        # @java return items;
        return items.sort(lambda a, b: key_selector(a) - key_selector(b))
    
    @classmethod
    def remove_last_n(cls, items, count):
        # @java items.subList(items.size() - count, items.size()).clear();
        items.splice(len(items) - count, count)

File: Java/src/main/java/OneLang/Utils/ArrayHelper/ArrayHelper.java
package OneLang.Utils.ArrayHelper;



import java.util.Arrays;
import java.util.function.Function;
import java.util.List;

public class ArrayHelper {
    public static <T> T[] sortBy(T[] items, Function<T, Integer> keySelector) {
        // @java-import java.util.Arrays
        // @java Arrays.sort(items, (a, b) -> keySelector.apply(a) - keySelector.apply(b));
        // @java return items;
        Arrays.sort(items, (a, b) -> keySelector.apply(a) - keySelector.apply(b));
        return items;
    }
    
    public static <T> void removeLastN(List<T> items, Integer count) {
        // @java items.subList(items.size() - count, items.size()).clear();
        items.subList(items.size() - count, items.size()).clear();
    }
}